Automattic/jetpack
# TreeSelector: Add tree selector component.

Fixes #

**Note: The sample code will be removed before merging `newsletter/subscription-settings.jsx` and `mock-data.jsx` will be removed before merging.**

## Proposed changes:

* Add general purpose tree selector component.
* It expects an array with `ID`, `parent`, and `name`.
* It triggers the event `onSelect`, `onDeselect` and `onChange`.
* It can search using the `keyword` property.
* It will show a flat list with breadcrumb when searching.

### Other information:


## Jetpack product discussion
This is part of the Newsletter Categories in Jetpack i2. p1HpG7-rd9-p2




## Does this pull request change what data or activity we track or use?



## Testing instructions:
* Go to `/wp-admin/admin.php?page=jetpack#/newsletter`.
* You will see a sample tree at the bottom of the page.
* Inspect the console logs while you select and deselect the tree items, you should see event logs.

## Screenshot

https://github.com/Automattic/jetpack/assets/1287077/f3a53c1e-7e46-4880-8852-6e5049471d51

![2024-02-20_20-44-26](https://github.com/Automattic/jetpack/assets/1287077/33dc0336-62c0-423c-8224-bcea26cfeb65)
